ShelterModule/Models/Announcements/GetAllAnnouncementsQuery.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace ShelterModule.Models.Announcements;
+
+public sealed class GetAllAnnouncementsFilteredQuery
+{
+    public string[]? Species { get; init; } = null!;
+    public string[]? Breeds { get; init; } = null!;
+    public string[]? Cities { get; init; } = null!;
+    public int? MinAge { get; init; }
+    public int? MaxAge { get; init; }
+    public string[]? ShelterNames { get; init; } = null!;
```

Comment:
We could have some less derived type here than an array, like, let's say, `IReadOnlyList<>`. This will ensure that we won't be able to accidentally modify collections that we receive in request, and will allow us to create this properties (in unit tests) in a more convenient way since we will be able to pass any list-like collection, not just an array. 

If for some reason we can't bind query lists to readonly collections (because deserializers/binders have some quirks like that sometimes), we can use `IList<>` instead.